src/dune_rules/jsoo_rules.ml

```diff
@@ -72,6 +72,53 @@ end = struct
       | name, false -> [ "--disable"; name ])
 end
 
+module Version = struct
+  type t = int list
+
+  let of_string s : t =
+    let s =
+      match
+        String.findi s ~f:(function
+          | '+' | '-' | '~' -> true
+          | _ -> false)
+      with
+      | None -> s
+      | Some i -> String.take s i
+    in
+    String.split s ~on:'.' |> List.map ~f:int_of_string
+
+  let rec compare v v' =
+    match (v, v') with
+    | [ x ], [ y ] -> Int.compare x y
+    | [], [] -> Ordering.Eq
+    | [], y :: _ -> Int.compare 0 y
+    | x :: _, [] -> Int.compare x 0
+    | x :: xs, y :: ys -> (
+      match Int.compare x y with
+      | Eq -> compare xs ys
+      | n -> n)
+
+  let impl_version bin =
+    let open Memo.O in
+    let* _ = Build_system.build_file bin in
+    Memo.of_reproducible_fiber
+    @@ Process.run_capture_line Process.Strict bin [ "--version" ]
+    |> Memo.map ~f:(fun s -> try Some (of_string s) with _ -> None)
+
+  let version_memo =
+    Memo.create "jsoo-version" ~input:(module Path) impl_version
+
+  let jsoo_verion path =
+    let open Memo.O in
+    let* jsoo = path in
+    let+ jsoo_verion =
+      match jsoo with
+      | Ok jsoo_path -> Memo.exec version_memo jsoo_path
+      | Error e -> Action.Prog.Not_found.raise e
+    in
+    jsoo_verion
```

Comment:
Don't think this temporary variable is needed.

Reply:
removed